langlib/lang.float/src/main/java/org/ballerinalang/langlib/floatingpoint/ToExpString.java

```diff
@@ -0,0 +1,110 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.langlib.floatingpoint;
+
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.internal.ErrorUtils;
+import io.ballerina.runtime.internal.FloatUtils;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+
+/**
+ * Native implementation of lang.float:toExpString(float, int?).
+ *
+ * @since 2201.1.0
+ */
+public class ToExpString {
+
+    public static BString toExpString(double x, Object fractionDigits) {
+        // If `x` is NaN or infinite, the result will be the same as `value:toString`.
+        BString str = FloatUtils.getBStringIfInfiniteOrNaN(x);
+        if (str != null) {
+            return str;
+        }
+
+        long noOfFractionDigits;
+
+        // If fractionalDigits is `()`, use the minimum number of digits required to accurately represent the value.
+        double xAbsValue = Math.abs(x);
+        if (fractionDigits == null) {
+            if (xAbsValue == 0) {
+                noOfFractionDigits = 1;
+            } else {
+                int integerPart = (int) (Math.log10(xAbsValue));
+                noOfFractionDigits = BigDecimal.valueOf(xAbsValue / Math.pow(10, integerPart)).scale();
+            }
+        } else {
+            noOfFractionDigits = (long) fractionDigits;
+        }
+
+        // Panic if noOfFractionDigits < 0.
+        if (noOfFractionDigits < 0) {
+            throw ErrorUtils.createInvalidFractionDigitsError();
```

Comment:
Shall we add tests for error throwing

Reply:
Lets address them here: https://github.com/ballerina-platform/ballerina-lang/issues/35757

Reply:
i have already added tests